### PR TITLE
Update 4.0

### PR DESCRIPTION
**Pull request by bbhtt**
Runtime 43 is needed for newer glib 2.74 required by glibmm. I think the issue with openssl 3.0 got resolved since I see https://github.com/transmission/transmission/issues/1777

Switched all gtkmm dependencies to meson and added cleanups.

This built locally but I didn't run it.

Closes https://github.com/flathub/com.transmissionbt.Transmission/issues/54, closes https://github.com/flathub/com.transmissionbt.Transmission/issues/46

### CODE REVIEW

**Comment by bbhtt** [comment]
Getting a `Failed to register: GDBus.Error:org.freedesktop.DBus.Error.ServiceUnknown: org.freedesktop.DBus.Error.ServiceUnknown` when trying to run. Probably some access is needed because it works with

`flatpak run --socket=session-bus  com.transmissionbt.Transmission`

**Comment by bbhtt** [comment]
Looks like they don't have a fixed application id: https://github.com/transmission/transmission/blob/1f1e614f9975399f5c5e847c76f4f61104649726/gtk/Application.cc#L703

eg. `com.transmissionbt.transmission_65024_34103683`

Needs `--own-name=com.transmissionbt.* `,

**Comment by bbhtt** [comment]
> Build [26226 successful](https://buildbot.flathub.org/#/builders/6/builds/26226) To test this build, install it from the testing repository:
> 
> ```
> flatpak install --user https://dl.flathub.org/build-repo/8885/com.transmissionbt.Transmission.flatpakref
> ```

Seems to be working fine now

File: com.transmissionbt.Transmission.json [hunk 1]
```diff
@@ -18,14 +19,18 @@
         "--talk-name=org.freedesktop.Notifications",
         "--filesystem=xdg-run/gvfsd",
         "--talk-name=org.gtk.vfs.*",
-        "--talk-name=org.kde.StatusNotifierWatcher"
+        "--talk-name=org.kde.StatusNotifierWatcher",
+        "--own-name=com.transmissionbt.*"
```
Comment (wjt) [inline com.transmissionbt.Transmission.json:23]: I previously had a patch to make this unnecessary. It was dropped in this update. Why? It should be pretty easy to refresh it.

**Review by wjt — COMMENTED** [review]
Nice work! I just have one question:

File: com.transmissionbt.Transmission.json (hunk 1 above)
Comment (bbhtt) [reply com.transmissionbt.Transmission.json:23]: The file (gtk/main.c) doesn't exist anymore. I can recreate that if you want.

**Review by wjt — APPROVED** [review]

**Comment by wjt** [comment]
Thanks. Would you like commit access to this repo?

**Comment by Amanse** [comment]
nice lessgoooo

**Comment by bbhtt** [comment]
> Thanks. Would you like commit access to this repo?

No, but thanks. I'll try to send PRs if I can.

**Pull request merged**